optimizely/optimizely.py

```diff
@@ -287,6 +287,96 @@ def _get_feature_variable_for_type(
+    """
+        if not validator.is_non_empty_string(feature_key):
+            self.logger.error(enums.Errors.INVALID_INPUT.format('feature_key'))
+            return None
+
+        if not isinstance(user_id, string_types):
+            self.logger.error(enums.Errors.INVALID_INPUT.format('user_id'))
+            return None
+
+        if not self._validate_user_inputs(attributes):
+            return None
+
+        feature_flag = project_config.get_feature_from_key(feature_key)
+        if not feature_flag:
+            return None
+
+        feature_enabled = False
+        source_info = {}
+
+        decision = self.decision_service.get_variation_for_feature(project_config, feature_flag, user_id, attributes)
+        if decision.variation:
+
+            feature_enabled = decision.variation.featureEnabled
+            if feature_enabled:
+                self.logger.info(
+                    'Feature "%s" for variation "%s" is enabled.' % (feature_key, decision.variation.key)
+                )
+            else:
+                self.logger.info(
+                    'Feature "%s" for variation "%s" is not enabled.' % (feature_key, decision.variation.key)
+                )
+        else:
+            self.logger.info(
+                'User "%s" is not in any variation or rollout rule. '
+                'Returning default value for all variables of feature flag "%s".' % (user_id, feature_key)
+            )
+
+        all_variables = {}
+        for variable_key in feature_flag.variables:
+            variable = project_config.get_variable_for_feature(feature_key, variable_key)
+            variable_value = variable.defaultValue
+            if feature_enabled:
+                variable_value = project_config.get_variable_value_for_variation(variable, decision.variation)
+                self.logger.info(
+                    'Got variable value "%s" for variable "%s" of feature flag "%s".'
+                    % (variable_value, variable_key, feature_key)
+                )
+
+            try:
+                actual_value = project_config.get_typecast_value(variable_value, variable.type)
+            except:
+                self.logger.error('Unable to cast value. Returning None.')
+                actual_value = None
+
+            all_variables[variable_key] = actual_value
+
+        if decision.source == enums.DecisionSources.FEATURE_TEST:
+            source_info = {
+                'experiment_key': decision.experiment.key,
+                'variation_key': decision.variation.key,
+            }
+
+        self.notification_center.send_notifications(
+            enums.NotificationTypes.DECISION,
+            enums.DecisionNotificationTypes.FEATURE,
+            user_id,
+            attributes or {},
+            {
+                'feature_key': feature_key,
+                'feature_enabled': feature_enabled,
+                'source': decision.source,
+                'source_info': source_info,
+            },
+        )
```

Comment:
This doesn't make sense. I think we should trigger multiple `feature-variable` notifications: https://docs.developers.optimizely.com/full-stack/docs/set-up-notification-listener-swift#section-set-up-a-decision-notification-listener

We do the same thing in `get_enabled_features` as well.

Reply:
the idea is to return only one notification in get_all_feature_variables, and not multiple notifications.